packages/mdc-checkbox/foundation.ts

```diff
@@ -98,7 +101,19 @@ export class MDCCheckboxFoundation extends MDCFoundation<MDCCheckboxAdapter> {
   }
 
   /**
-   * Handles the change event for the checkbox
+   * Handles the click event for the checkbox
+   * added for IE browser to fix compatibility issue b/144588184
+   */
+  handleClick() {
+    const {TRANSITION_STATE_INDETERMINATE} = strings;
+    if (this.currentCheckState_ === TRANSITION_STATE_INDETERMINATE) {
+      this.adapter_.setChecked(!this.previousIsChecked_);
+    }
+    this.handleChange();
```

Comment:
It is best not to call other event handlers within event handlers. So here you can call `transitionCheckState_` directly